### PR TITLE
WFLY-13854 WildFly cannot connect to an infinispan-server cluster req…

### PR DESCRIPTION
**Pull request by rhusar**
…uiring authentication using properties

Resolves
https://issues.redhat.com/browse/WFLY-13854

### CODE REVIEW

**Comment by rhusar** [comment]
@pferraro For review.

**Comment by pferraro** [comment]
@rhusar FYI - there are regressions in the *HotRodWebFailoverTestCase tests.

**Comment by rhusar** [comment]
@pferraro I know – it looks like the statistics are not counted with authenticated accesses, which seems very odd. I have been investigating with no luck so far. The workaround is already in place, see {{org.jboss.as.test.clustering.cluster.web.remote.AbstractHotRodWebFailoverTestCase#testGracefulSimpleFailover}}.

**Comment by rhusar** [comment]
Thanks @pferraro for finding the issue with how infinispan handles {{org.infinispan.client.hotrod.configuration.ConfigurationBuilder#withProperties}} which fixed the issues above.

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteCacheContainerResourceDefinition.java [hunk 1]
```diff
@@ -147,6 +156,12 @@ public static void buildTransformation(ModelVersion version, ResourceTransformat
         } else {
             ResourceTransformationDescriptionBuilder builder = parent.addChildResource(RemoteCacheContainerResourceDefinition.WILDCARD_PATH);
 
+            if (InfinispanModel.VERSION_13_0_0.requiresTransformation(version)) {
+                builder.getAttributeBuilder()
+                        .setDiscard(new DiscardAttributeValueChecker(false, true, ModelNode.FALSE), Attribute.PROPERTIES.getDefinition())
```
Comment (pferraro) [inline clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteCacheContainerResourceDefinition.java:161]: Given that this is a map attribute, when is this value ever FALSE?
Why not use DiscarcAttributeChecker.UNDEFINED?

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteCacheContainerConfigurationServiceConfigurator.java [hunk 2]
```diff
@@ -135,6 +140,11 @@ public ServiceConfigurator configure(OperationContext context, ModelNode model)
 
         this.server = context.hasOptionalCapability(CommonRequirement.MBEAN_SERVER.getName(), null, null) ? new ServiceSupplierDependency<>(CommonRequirement.MBEAN_SERVER.getServiceName(context)) : null;
 
+        this.properties.clear();
+        for (Property property : ModelNodes.optionalPropertyList(Attribute.PROPERTIES.resolveModelAttribute(context, model)).orElse(Collections.emptyList())) {
```
Comment (pferraro) [inline clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteCacheContainerConfigurationServiceConfigurator.java:144]: N.B. Once we upgrade to jboss-dmr 1.6.x, we can replace this with:
```java
for (Property property : Attribute.PROPERTIES.resolveModelAttribute(context, model).asPropertyListOrEmpty())
```

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteCacheContainerResourceDefinition.java (hunk 1 above)
Comment (rhusar) [reply clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteCacheContainerResourceDefinition.java]: Thanks for spotting this, fixed + fixed the corresponding rejection test.

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteCacheContainerConfigurationServiceConfigurator.java (hunk 2 above)
Comment (rhusar) [reply clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteCacheContainerConfigurationServiceConfigurator.java:144]: Thanks - noted and put on a list.

**Review by pferraro — APPROVED** [review]

**Pull request merged**